src/components/Shared/AddressTab.js

```diff
@@ -208,98 +209,69 @@ const AddressTab = ({
+
+             valueField='name'
+             displayField='name'
+             name='cityId'
+             label={labels.city}
+             required={requiredOptional ? false : true}
+             readOnly={(readOnly || !addressValidation.values.countryId) && true}
+             form={addressValidation}
+             secondDisplayField={false}
+             onChange={(event, newValue) => {
+              if (newValue) {
+                addressValidation.setFieldValue('cityId', newValue?.recordId)
+                addressValidation.setFieldValue('city', newValue?.name)
+              } else {
+                addressValidation.setFieldValue('cityId', null)
+                addressValidation.setFieldValue('city', null)
+              }
+              addressValidation.setFieldValue('cityDistrictId', null)
+              addressValidation.setFieldValue('cityDistrict', null)
+            }}
+
             />
           </Grid>
 
-          <Grid item xs={12}>
-            <CustomLookup
-              name='cityDistrict'
-              label={labels.cityDistrict}
-              valueField='name'
-              displayField='name'
-              readOnly={(readOnly || !addressValidation.values.cityId) && true}
-              store={cityDistrictStore}
-              setStore={setCityDistrictStore}
-              onLookup={searchQry => {
-                setCityDistrictStore([])
-                var parameters = `_size=30&_startAt=0&_filter=${searchQry}&_cityId=${addressValidation.values.cityId}`
+           <Grid item xs={12}>
 
-                getRequest({
-                  extension: SystemRepository.CityDistrict.snapshot,
-                  parameters: parameters
-                })
-                  .then(res => {
-                    console.log(res.list)
-                    setCityDistrictStore(res.list)
-                  })
-                  .catch(error => {
-                    setErrorMessage(error)
-                  })
-              }}
-              firstValue={addressValidation.values.cityDistrict}
-              secondDisplayField={false}
-              onChange={(event, newValue) => {
-                if (newValue) {
-                  addressValidation.setFieldValue('cityDistrictId', newValue?.recordId)
-                  addressValidation.setFieldValue('cityDistrict', newValue?.name)
-                } else {
-                  addressValidation.setFieldValue('cityDistrictId', null)
-                  addressValidation.setFieldValue('cityDistrict', null)
-                }
-              }}
-              error={addressValidation.touched.cityDistrictId && Boolean(addressValidation.errors.cityDistrictId)}
-              helperText={addressValidation.touched.cityDistrictId && addressValidation.errors.cityDistrictId}
-              maxAccess={maxAccess}
+           <ResourceLookup
+             endpointId={SystemRepository.CityDistrict.snapshot}
+             parameters={{
+              _size: 30,
+              _startAt: 0,
```

Comment:
SY.snapshotCDI requests only _cityId and _filter
please remove the unnecessary parameters